src/main/java/org/apache/sysml/runtime/matrix/data/LibMatrixCUDA.java

```diff
@@ -948,4 +983,173 @@ public static boolean isInSparseFormat(MatrixObject mo) {
 			return mo.getGPUObject().isInSparseFormat();
 		return MatrixBlock.evalSparseFormatInMemory(mo.getNumRows(), mo.getNumColumns(), mo.getNnz());
 	}
+	
+	/**
+	 * Performs elementwise multiplication of input vector in and constant using cublasDaxpy
+	 * 
+	 * @param ec
+	 * @param in
+	 * @param constant
+	 * @param outputName
+	 * @param op
+	 * @throws DMLRuntimeException
+	 */
+	public static void vectorScalarMult(ExecutionContext ec, MatrixObject in, double constant, String outputName, GPUEnabledElementwiseOp op) throws DMLRuntimeException {
```

Comment:
This is a good idea. The GPU matrix libraries will resemble the CP matrix libraries. Much like how LibMatrixCUDA (for now) looks like LibMatrixCUDA.
However, there is nothing stopping us from making all the functions well named and well engineered :)